TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php

```diff
@@ -145,70 +146,135 @@ public function setName($newName) {
 	 *
 	 * @param string $path
 	 * @param boolean $checkForExistence Checks for existence at target path, internally used for recursions and shadow nodes.
+	 * @return void
 	 * @throws NodeException
-	 * @internal This method is purely internal and Node objects can call this on other Node objects
 	 */
-	public function setPath($path, $checkForExistence = TRUE) {
+	protected function setPath($path, $checkForExistence = TRUE) {
```

Comment:
That's the trick, it is absolutely valid to call a protected method on a instance of the same class other than $this. And it is not called from anywhere else outside of Node.